src/util/xkcd.py

```diff
@@ -0,0 +1,67 @@
+"""Copyright by CookieDecide, Darkuuu
+Licensed under MIT License
+Contains various helper functions and classes to receive content from xkcd.
+"""
+
+import urllib.request
+import json
+import random
+
+_url_comic_current = "https://xkcd.com/info.0.json"
+_url_comic_left = "https://xkcd.com/"
+_url_comic_right = "/info.0.json"
+
+
+class XkcdComic:
+    """Represents a comic from xkcd.com. Currently only has title and image URL,
+    might be changed in future.
+
+    Attributes:
+        title: Title of the comic.
+        img_url: URL of the image of the comic.
+        number: Number of the comic on website archive.
+        url: URL pointing to comic (can be derived from comic number).
+    """
+
+    def __init__(self, title, img_url, number):
+        self.title = title
+        self.img_url = img_url
+        self.number = number
+        self.url = _url_comic_left + str(number) + "/"
+
+
+def get_random_xkcd() -> XkcdComic:
+    """Fetches a random comic from xkcd.com.
+    API is described at https://xkcd.com/json.html
+
+    Returns:
+        XkcdComic: Valid comic object.
+        None: Only returned when not able to fetch comic.
+    """
+    max_number = 0
+    try:
+        # Get current comic to derive maximum amount of available comics
+        with urllib.request.urlopen(_url_comic_current) as url:
+            data = json.load(url)
+            max_number = data["num"]
+            rand_number = random.randint(1, max_number)
+            return create_xkcdcomic(rand_number)
+    except Exception as err:
```

Comment:
It is not good practice to catch a generic exception, try to use specific exceptions that can occure here or at leat log the exception.

Reply:
Fixed, catching now HTTP error and log it if something goes wrong here. Anything else should be considered a bug if happening.